src/types/UiaTextRangeBase.cpp

```diff
@@ -985,29 +985,15 @@ std::wstring UiaTextRangeBase::_getTextValue(til::CoordType maxLength) const
         auto inclusiveEnd = _end;
         bufferSize.DecrementInBounds(inclusiveEnd, true);
 
-        // reserve size in accordance to extracted text
-        const auto textRects = buffer.GetTextRects(_start, inclusiveEnd, _blockRange, true);
-        const auto bufferData = buffer.GetText(true,
-                                               false,
-                                               textRects);
-        const size_t textDataSize = bufferData.text.size() * bufferSize.Width();
-        textData.reserve(textDataSize);
-        for (const auto& text : bufferData.text)
-        {
-            if (textData.size() >= maxLengthAsSize)
-            {
-                // early exit; we're already at/past max length
-                break;
-            }
-            textData += text;
-        }
+        const auto req = TextBuffer::CopyRequest{ buffer, _start, inclusiveEnd, _blockRange, true, false, false, true };

```

Comment:
I did try that before but it didn't work. The ctor does all the range checks for us before init the members (and it takes a ref to buffer for that reason).

Is there any other way we can do this without repeating the ctor logic here?

Reply:
Ah right I forgot to delete this comment after reading the constructor definition later on.
You could do the clamping within the 3 functions. The cost should be in the ballpark of only <20ns after all. But I don't think we need to make such a modification now because this is fine as well.